src/app/components/SwitchChain.tsx

```diff
@@ -54,8 +57,10 @@ export const SwitchChain = () => {
       } finally {
         triggerLoading(false);
       }
+
+      await getChainId();
```

Comment:
Why do we need to getChainId again after switching chains?
If the switchChain is successful, update selectedChain = chain.chainId; 
Else, nothing changes

Reply:
Ah I see, after testing again, it's sure that we dont need get chain again here, the global chainId from walletconnect automatically update when chain changed.